api/api/cat.js

```diff
@@ -214,6 +214,27 @@ CatApi.prototype.indices = function catIndicesApi (params, options, callback) {
   return this.transport.request(request, options, callback)
 }
 
+CatApi.prototype.cluster_manager = function catClusterManagerApi (params, options, callback) {
+  ;[params, options, callback] = normalizeArguments(params, options, callback)
+
+  let { method, body, ...querystring } = params
+  querystring = snakeCaseKeys(acceptedQuerystring, snakeCase, querystring)
+
+  let path = ''
+  if (method == null) method = 'GET'
+  path = '/' + '_cat' + '/' + 'cluster_manager'
+
+  // build request object
+  const request = {
+    method,
+    path,
+    body: null,
+    querystring
+  }
+
+  return this.transport.request(request, options, callback)
+}
+
 CatApi.prototype.master = function catMasterApi (params, options, callback) {
```

Comment:
Josh, I don't quite understand this comment. `catClusterManagerApi` uses path `/_cat/cluster_manager` and `catMasterApi` uses path `/_cat/cluster_manager`. Both paths are supported in OpenSearch til 3.0 (here is [the OS PR](https://github.com/opensearch-project/OpenSearch/pull/2404/files) ). If we do `CatApi.prototype.master = catClusterManagerApi` to only support `/_cat/cluster_manager` then would that cause breaking changes? and how could we support `/_cat/master`? Or you are suggesting a cleaner way to reduce duplication? if so, pls be more specific. Thank you.

Reply:
I previously thought I could wrap the original function and make it to accept a path. Something like, `CatApi.prototype.master = catPathApi (master_path)` and `CatApi.prototype.cluster_manager = catPathApi (cluster_manager_path)`. Then I thought in 3.0, instead of simply removing all the master functions, we need to modify it back because we don't have to define a variable then pass it. Do you have some good suggestions?

Reply:
Hm, that's a good point, that it potentially makes the cleanup and removal more straightforward. I think it's a fine trade-off decision, but I'd probably at least add a comment that mentions that any changes made inside these should be duplicated in both.

Reply:
I will add some comments to all the changes, including the explains and future actions.